types/comtac/Cluey/uplink.js

```diff
@@ -1723,40 +2544,152 @@ function consume(event) {
-    digitalInputs.limit = digital.cot.limit;
-    digitalInputs.event = digital.cot.event;
-    digitalInputs.interrogation = digital.cot.interrogation;
-    digitalInputs.cyclic = digital.cot.cyclic;
-
-    digitalInputs.blocked = digital.status.blocked;
-    digitalInputs.state = digital.status.state;
-
-    const timestamp = new Date(digital.timestamp.string);
-
-    emit("sample", {
-      data: digitalInputs,
-      topic: "digital",
-      timestamp,
-    });
+  const { errors } = decoded;
+
+  switch (type) {
+    case "FIXED_DATA":
+      delete data.timestamp;
+      Object.keys(data).forEach((key) => {
+        data[key].limit = data[key].cot.limit;
+        data[key].event = data[key].cot.event;
+        data[key].interrogation = data[key].cot.interrogation;
+        data[key].cyclic = data[key].cot.cyclic;
+        delete data[key].cot;
+
+        emit("sample", { data: data[key], topic: key });
+      });
+      break;
+    case "DI_DATA":
+      data.digitalInputs.forEach((dataPoint) => {
+        let result = {};
+        result.limit = dataPoint.cot.limit;
+        result.event = dataPoint.cot.event;
+        result.interrogation = dataPoint.cot.interrogation;
+        result.cyclic = dataPoint.cot.cyclic;
+        delete dataPoint.cot;
+
+        // Timestamp
+        const timestamp = checkTimestamp(
+          new Date(dataPoint.timestamp.unix * 1000),
+        );
+        delete dataPoint.timestamp;
+
+        result = Object.assign(result, dataPoint);
+
+        emit("sample", {
+          data: result,
+          topic: "point_info",
+          timestamp,
+        });
+      });
+      break;
+    case "AI_DATA":
+      data.analogInputs.forEach((dataPoint) => {
+        const { topic } = dataPoint;
+        const timestamp = checkTimestamp(
+          new Date(dataPoint.timestamp.unix * 1000),
+        );
+
+        const result = {};
+        result.limit = dataPoint.cot.limit;
+        result.event = dataPoint.cot.event;
+        result.interrogation = dataPoint.cot.interrogation;
+        result.cyclic = dataPoint.cot.cyclic;
+
+        delete dataPoint.cot;
+        delete dataPoint.topic;
+        delete dataPoint.timestamp;
+
+        dataPoint = Object.assign(result, dataPoint);
+
+        emit("sample", { data: result, topic, timestamp });
+      });
+      break;
+    case "CNT_DATA":
+      data.digitalInputs.forEach((dataPoint) => {
+        const result = {};
+        result.limit = dataPoint.cot.limit;
+        result.event = dataPoint.cot.event;
+        result.interrogation = dataPoint.cot.interrogation;
+        result.cyclic = dataPoint.cot.cyclic;
+        delete dataPoint.cot;
+
+        // Timestamp
+        const timestamp = checkTimestamp(
+          new Date(dataPoint.timestamp.unix * 1000),
+        );
+        delete dataPoint.timestamp;
+
+        dataPoint = Object.assign(result, dataPoint);
+
+        emit("sample", { data: result, topic: "count_data", timestamp });
```

Comment:
map the output explicitly as proposed above rather than assigning and deleting values